### ISSUE TITLE
Handle empty fields

### ISSUE THREAD

**Issue by cassc** [testdouble/clojurescript.csv#10]
It looks like `read-csv` is not handling empty field correctly. 

`(read-csv (write-csv [["a" "b" "" "d"]]))`

returns 

`[["a" "b" ",d"]]`

**Comment by jeremyrsellars** [comment]
The problem may relate to consecutive delimiters.
It doesn't work with
```clojure
cljs.user=> (testdouble.cljs.csv/read-csv "a,b,c\n1,,")
[["a" "b" "c"] ["1" ","]]
```
But it works fine with quoted empty fields.

```clojure
cljs.user=> (testdouble.cljs.csv/read-csv "a,b,c\n1,\"\",")
[["a" "b" "c"] ["1" "" ""]]
```

**Comment by rosston** [comment]
Thanks for the added detail! I'll try to take a look at this soon and see if I can sketch out a fix.

**Comment by cassc** [comment]
I added a possible fix [here](https://github.com/cassc/clojurescript.csv/commit/294827e90cff61f6b3bd266c9e9a91ceeb5db297#diff-188c6328c8104a2cfe3a6bf4047b3d97L139)

**Comment by rosston** [comment]
That fix seems plausible! I would welcome a PR with a failing test around empty fields and eventual fix (with existing tests passing, of course), if you're up for it.

**Comment by rosston** [comment]
Fix released in v0.4.1. Thanks @cassc!
